packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/repository.ts

```diff
@@ -524,45 +532,59 @@ export class SavedObjectsRepository implements ISavedObjectsRepository {
+    //   types: new Set(typesAndSpaces.keys()),
+    //   spaces: spacesToAuthorize,
+    //   enforceMap: typesAndSpaces,
+    //   options: { allowGlobalResource: true },
+    //   auditOptions: {
+    //     objects: validObjects.map((element) => {
+    //       return {
+    //         type: element.value.object.type,
+    //         id: element.value.object.id,
+    //       };
+    //     }),
+    //   },
+    // });
+
+    const authObjects: AuthorizeCreateObject[] = validObjects.map((element) => {
+      const { object, preflightCheckIndex: index } = element.value;
       const preflightResult = index !== undefined ? preflightCheckResponse[index] : undefined;
+      return {
+        type: object.type,
+        id: object.id,
+        initialNamespaces: object.initialNamespaces,
+        existingNamespaces: preflightResult?.existingDocument?._source.namespaces,
+      };
+    });
 
-      const spacesToEnforce = typesAndSpaces.get(object.type) ?? new Set([namespaceString]); // Always enforce authZ for the active space
-      for (const space of object.initialNamespaces ?? []) {
-        spacesToEnforce.add(space);
-        spacesToAuthorize.add(space);
-      }
-      typesAndSpaces.set(object.type, spacesToEnforce);
-      for (const space of preflightResult?.existingDocument?._source.namespaces ?? []) {
-        if (space === ALL_NAMESPACES_STRING) continue; // Don't accidentally check for global privileges when the object exists in '*'
-        spacesToAuthorize.add(space); // existing namespaces are included so we can later redact if necessary
-      }
-    }
-
-    const authorizationResult = await this._securityExtension?.performAuthorization({
-      // If a user tries to create an object with `initialNamespaces: ['*']`, they need to have 'bulk_create' privileges for the Global
-      // Resource (e.g., All privileges for All Spaces).
-      // Inversely, if a user tries to overwrite an object that already exists in '*', they don't need to have 'bulk_create' privileges for the Global
-      // Resource, so in that case we have to filter out that string from spacesToAuthorize (because `allowGlobalResource: true` is used
-      // below.)
-      actions: new Set(['bulk_create']),
-      types: new Set(typesAndSpaces.keys()),
-      spaces: spacesToAuthorize,
-      enforceMap: typesAndSpaces,
-      auditCallback: (error) => {
-        for (const { value } of validObjects) {
-          this._securityExtension!.addAuditEvent({
-            action: AuditAction.CREATE,
-            savedObject: { type: value.object.type, id: value.object.id },
-            error,
-            ...(!error && { outcome: 'unknown' }), // If authorization was a success, the outcome is unknown because the create operation has not occurred yet
-          });
-        }
-      },
-      options: { allowGlobalResource: true },
+    const authorizationResult = await this._securityExtension?.authorizeCreate({
```

Comment:
The above commented code is replaced by this extension call, which handles figuring out how to authorize the create action based on the the preflight results.